File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/SmsForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SmsForm {
    ApiKey: StringEditor;
    Sender: StringEditor;
    EntityId: StringEditor;
    TemplateId: StringEditor;
}

export class SmsForm extends PrefixedContext {
    static readonly formKey = 'Settings.Sms';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SmsForm.init)  {
            SmsForm.init = true;

            var w0 = StringEditor;

            initFormType(SmsForm, [
                'ApiKey', w0,
                'Sender', w0,
                'EntityId', w0,
                'TemplateId', w0
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Settings/Sms/SmsForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Settings.Forms;

[FormScript("Settings.Sms")]
[BasedOnRow(typeof(SmsRow), CheckNames = true)]
public class SmsForm
{
    public string ApiKey { get; set; }
    public string Sender { get; set; }
    public string EntityId { get; set; }
    public string TemplateId { get; set; }
}